product-catalog/app/productcatalog/pages/FacetData.java

```diff
@@ -0,0 +1,38 @@
+package productcatalog.pages;
+
+import io.sphere.sdk.facets.Facet;
+import io.sphere.sdk.models.Base;
+
+import static io.sphere.sdk.facets.DefaultFacetType.*;
+
+public class FacetData extends Base {
+    private Facet facet;
+
+    public FacetData(final Facet<?> facet) {
+        this.facet = facet;
+    }
+
+    public Facet getFacet() {
+        return facet;
+    }
+
+    public boolean isSortedSelectFacet() {
+        return facet.getType().equals(SORTED_SELECT);
```

Comment:
Here you could extract a private method for all of them which takes the type as parameter.